### ISSUE TITLE
Dashboards: Variable query doesn't support Prometheus `or` operation

### ISSUE THREAD

**Issue by superlevure** [grafana/grafana#93972]
### What happened?

Using the following query in a variable template:

```
kube_pod_labels{my_first_label=~"value"} or kube_pod_labels{my_second_label=~"other_value"}
```

results in: `parse error: unexpected <op:or>`

### What did you expect to happen?

Returns the labels of the pods that match `my_first_label=~"value"` **or** `my_second_label=~"other_value"`

### Did this work before?

No, but the same query does work using the explorer UI

### How do we reproduce it?

1. Create a new dashboard
2. Create a new variable
3. Select Prometheus as the variable data source
4. Use `kube_pod_labels{my_first_label=~"value"} or kube_pod_labels{my_second_label=~"other_value"}` as the query

### Environment (with versions)?

Grafana: v9.4.17
OS: macOS Sonoma
Browser: Firefox 130.0

### Grafana platform?

Kubernetes

### Datasource(s)?

Prometheus

**Comment by DelusionArun** [comment]
The promQL sample in the grafana playground does appear to use OR expression successfully.

Home > Dashboards > Demo: Prometheus > kube-state-metrics - Home
![Image](https://github.com/user-attachments/assets/998ea6bf-37f4-465c-902d-da3b1337b95b)

I wonder if the issue could be something different?

**Comment by superlevure** [comment]
Indeed, as wrote in the issue the issue is only for variable template queries, it does work fine when querying the datasource in Explore or in Dashboard's Panel

**Comment by bohandley** [comment]
Hi @superlevure, it looks like v9.4.17 still has the old varialbe query editor that is a text box. For the syntax of using a query, can you try this?

`query_result(kube_pod_labels{my_first_label=~"value"} or kube_pod_labels{my_second_label=~"other_value")`

If you pass in a query without `query_result(<your query>)` it defaults to the series query which only accepts a single series without additional query syntax (e.g. go_goroutines{container:"something"}). The series type query would error as you are seeing.

[Here are docs for the variable query editor in v.9.4.17](https://github.com/grafana/grafana/blob/v9.4.x/docs/sources/datasources/prometheus/template-variables/index.md)

**Comment by superlevure** [comment]
Hi @bohandley, 

Thank you for your help! Indeed if I use `query_result(<my query>)` it works, but it looks like I can't apply `label_values` to it:

```
label_values(query_result(<my query>), my_label)
```

Gives me `invalid parameter "match[]": 1:13: parse error: unexpected "("`

Do you know if there's a way to achieve this?

**Comment by bohandley** [comment]
The variable query type for label values only supports a metric being passed in and not a series. So currently, there is no way to do this.

Can you make an issue for this a feature request? The label values endpoint in Prometheus can accept a `match[]` parameter and this is what we are passing the metric in as. We could also accept a series metric + labels, but then we would have to parse something like go_goroutines{label:"value"} into {__name__:"go_goroutines",label:"value"} as the new `match[]` param.

https://github.com/grafana/grafana/blob/938bc9c2ccc68e532935fc6e926cec06c8c2cb6e/packages/grafana-prometheus/src/metric_find_query.ts#L81-L112

**Comment by superlevure** [comment]
Thank you @bohandley, I've created https://github.com/grafana/grafana/issues/94289

**Comment by bohandley** [comment]
Thank you @superlevure! 

This will take a little planning to get to, as your use case is for the classic query still used in v9.4.17. One approach that would work in version > v11.x and the new query UI would be to support this in the new classic query option, which takes a function in the form of a string

I can't promise that I can get to it but our squad has taken a look at it. Thank you for your patience!

New UI here
![Image](https://github.com/user-attachments/assets/53c4f330-80d9-437d-a129-3c6c44179819)

**Comment by superlevure** [comment]
Thank you for your reactivity and your help @bohandley this is very much appreciated 🙏 I'll ping my team to see if we can update to v11.x

**Comment by itsmylife** [comment]
@superlevure I am closing this issue and tracking the feature request with https://github.com/grafana/grafana/issues/94289
